### ISSUE TITLE
page decode function parameter number error

### ISSUE THREAD

**Issue by HowcanoeWang** [cgohlke/tifffile#25]
Version: 2020.09.03
Environment: Anaconda, Python 3.6, Win10

For this decode example:
```python
>>> with TiffFile('my.tif') as tif:
...     fh = tif.filehandle
...     for page in tif.pages:
...         for index, (offset, bytecount) in enumerate(
...             zip(page.dataoffsets, page.databytecounts)
...         ):
...             fh.seek(offset)
...             data = fh.read(bytecount)
...             tile, indices, shape = page.decode(data, index,
...                                                page.jpegtables)
```
The last line `page.decode` give the following error:

`TypeError: decode() takes 2 positional arguments but 3 were given`

And I use the jupyter-notebook help function to check how the parameters required, it gives the function with only 2 parameters:
```
[in] ?page.decode
[out] Signature: page.decode(data, segmentindex)
```

Currently, I can simply ignore the `page.jpegtable` to make this function work, because for my case is `None`, but it conflict to the example and maybe some cases `jpgetable` is required, can you fix it or change the documents description?

**Comment by cgohlke** [comment]
Thank you. This will be fixed in the next version. I see how the current behavior is confusing/inconsistent although the example code is specifically chosen to "iterate over and decode single JPEG compressed tiles" and the `TiffPage.decode` `tables` parameter is documented "For JPEG compressed segments only".

**Comment by HowcanoeWang** [comment]
Thanks for your quick reply, btw, I tried my best to find the official documents about this package, but I found nothing except those example codes in readme.rst. Does it have any detailed official documentation website including functions api or others?

**Comment by cgohlke** [comment]
Should be fixed in v2020.9.22.
